nrf_security/src/zephyr/CMakeLists.txt

```diff
@@ -64,8 +64,13 @@ if(DEFINED src_zephyr)
     $<TARGET_PROPERTY:${mbedcrypto_target},INTERFACE_COMPILE_DEFINITIONS>
   )
 
-  zephyr_library_sources_ifdef(CONFIG_USERSPACE mbedtls_partition.c)
-  zephyr_library_app_memory(k_mbedtls_partition)
+  # When CONFIG_MBEDLTS is enabled Zephyr defines the MbedTLS
+  # partition in zephyr/kernel/userpace.c
+  if(NOT CONFIG_MBEDTLS)
+    zephyr_library_sources_ifdef(CONFIG_USERSPACE mbedtls_partition.c)
```

Comment:
I don't follow the logic here at all. If USERSPACE is enabled then userspace.c is compiled in, which defines a partition if CONFIG_MBEDTLS is enabled. So the partition is enabled if MBEDTLS && USERSPACE.
If we have a collision shouldn't it be if NOT (MEBDETLS && USERSPACE)?

NOT MBEDTLS && USERSPACE does not make sense to me.

Reply:
They had the ifdef with userspace, so my assumption is that this needs to be enabled only when userspace is enabled so  If I do :
if NOT (MBEDTLS && USERSPACE)

Then we will define this partition when:
CONFIG_MBEDTLS=y and CONFIG_USERSPACE=n

Isn't it?

Reply:
You talk about when it should not be added by us (MBEDTLS && USERSPACE). But when is it required?

Reply:
The comment says that Zephyr adds this partition, so why do we do so as well?

Reply:
@SebastianBoe  is the comment more clear now?

Reply:
Comment is so clear, that the bug is now obvious :)

Reply:
no bug, comment was unclear :( Clear now though.